hanadb_exporter/secret_manager.py

```diff
@@ -0,0 +1,32 @@
+import boto3
+import requests
+import json
+from botocore.exceptions import ClientError
+
+def get_db_credentials(secret_name):
+    ec2_info_url = "http://169.254.169.254/latest/dynamic/instance-identity/document"
+    ec2_info = requests.get(ec2_info_url).json()
+    region_name = ec2_info["region"]
+
+    # Create a Secrets Manager client
+    session = boto3.session.Session()
+    client = session.client(
+        service_name='secretsmanager',
+        region_name=region_name
+    )
+
+    # In this sample we only handle the specific exceptions for the 'GetSecretValue' API.
+    # See https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html
+    # We rethrow the exception by default.
+
+    try:
+        get_secret_value_response = client.get_secret_value(
+            SecretId=secret_name
+        )
+    except ClientError as e:
+        raise Exception("Couldn't retrieve secret details") from e
```

Comment:
Don't use that generic exceptions. Better to raise a some custom defined one

Reply:
Ok I am adding custom exception

Reply:
done https://github.com/SUSE/hanadb_exporter/pull/97